### ISSUE TITLE
Fix S1075 FN: Support collections and collection expressions

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#8172]
The rule should raise an issue when URIs or path delimiters are hardcoded as an element of a collection.
```cs
IList<string> uriList1 = new[] { "C:/test.txt" }; // FN
IList<string> uriList2 = ["C:/test.txt"]; // FN
string[][] uriMatrix1 = [["C:/test.txt"]]; // FN
IDictionary<string, string> uriDict = new Dictionary<string, string> { ["a"] = "C:/test.txt" }; // FN
```
